app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt

```diff
@@ -38,40 +40,112 @@ import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.painter.Painter
-import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import com.google.accompanist.permissions.ExperimentalPermissionsApi
 import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.permissions.shouldShowRationale
 import com.google.jetpackcamera.R
 
+/**
+ * Permission prompts screen.
+ * Camera permission will always prompt when disabled, and the app cannot be used otherwise
+ * if optional settings have not yet been declined by the user, then they will be prompted as well
+ */
 @OptIn(ExperimentalPermissionsApi::class)
 @Composable
-fun CameraPermission(cameraPermissionState: PermissionState, modifier: Modifier = Modifier) {
+fun PermissionsScreen(
+    modifier: Modifier = Modifier,
+    permissionEnums: Set<PermissionEnum>,
+    openAppSettings: () -> Unit
+) {
+    val permissionsViewModel =
```

Comment:
You should't create your ViewModel inside of the PermissionsScreen composable. You can do something similar to what we do in PreviewScreen with the PreviewViewmodel. Or use the `viewModel()` function : https://developer.android.com/reference/kotlin/androidx/lifecycle/viewmodel/compose/package-summary#viewmodel